### PR TITLE
Add bit extraction to concrete-core-ffi

### PR DESCRIPTION
**Pull request by agnesLeroy**
### Description
This PR adds bit extraction + some missing functions for it to concrete-core-ffi

### Checklist 

(Use '[x]' to check the checkboxes, or submit the PR and then click the checkboxes)

* [x] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [x] The PR description links to the related issue (to link an issue, use '#XXX'.)
* [x] The tests on AWS have been launched and are successful (comment with @slab-ci cpu_test and/or @slab-ci gpu_test to trigger the tests)
* [x] The draft release description has been updated
* [x] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by agnesLeroy** [comment]
@slab-ci cpu_test

**Comment by agnesLeroy** [comment]
@slab-ci cpu_test

**Review by IceTDrinker — COMMENTED** [review]
Thanks a lot for this, weren't new engine on views merged in another PR ? Otherwise generally looks good, I know there was a test failing so probably something to fix in concrete-core itself ?

File: concrete-core-ffi/src/backends/default/engines/glwe_secret_key_generation.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+/// Refer to `concrete-core` implementation for detailed documentation.
+///
+/// This function is [checked](crate#safety-checked-and-unchecked-functions).
+#[no_mangle]
+pub unsafe extern "C" fn default_engine_generate_new_glwe_secret_key_u64(
+    engine: *mut DefaultEngine,
+    dimension: usize,
+    poly_size: usize,
+    result: *mut *mut GlweSecretKey64,
+) -> c_int {
```
[the remark this answers is not in the document]
Comment (agnesLeroy) [reply concrete-core-ffi/src/backends/default/engines/glwe_secret_key_generation.rs:19]: Yeah I don't know in the end we'll just offer all entry points so I thought why not :sweat_smile:

File: concrete-core-ffi/tests/test_lwe_bit_extraction.c [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+  destroy_default_parallel_engine(default_parallel_engine);
+  destroy_default_engine(default_engine);
+  destroy_fftw_engine(fftw_engine);
+  destroy_seeder_builder(builder);
+  free(input_ct_buffer);
+  free(output_ct_vector_buffer);
+}
+
+//void bootstrap_unchecked_raw_ptr_buffers_test(void) {
+//  DefaultEngine *default_engine = NULL;
```
[the remark this answers is not in the document]
Comment (agnesLeroy) [reply concrete-core-ffi/tests/test_lwe_bit_extraction.c:120]: oh yes totally forgot to add the unchecked test! I'll do it!

**Comment by agnesLeroy** [comment]
@IceTDrinker it seems we were missing some entry points for views, so I just added them. Tests should be fixed now.

**Comment by IceTDrinker** [comment]
I think we can relaunch the tests to include the uncommented one, should cancel the previous slab run

**Comment by IceTDrinker** [comment]
@slab-ci cpu_test

**Review by IceTDrinker — APPROVED** [review]
Thanks looks good plus tests are green ✔️

**Pull request merged**